azavea/raster-foundry
# Receive scene metadata with uploads

## Overview

This PR adds two scene metadata fields to the uploads table -- cloud cover and acquisition date.
It parses them out separately from the incoming `metadata` json, throws them into a `sceneMetadata`
attribute, and passes all of the existing `metadata` in unchanged.

### Checklist

- ~Styleguide updated, if necessary~
- [x] Swagger specification updated, if necessary (if I had a dollar for every time this checkbox was the _only_ reason I updated the swagger spec...)

## Testing Instructions

This is your base test json:

```json
{"files":["ebee-test-flight-nz233reals.tif"],
 "datasource":"c14c8e97-ba85-4677-ac9c-069cfef1f0b1",
 "fileType":"GEOTIFF",
 "uploadStatus":"UPLOADING",
 "visibility":"PRIVATE",
 "organizationId":"dfac6307-b5ef-43f7-beda-b9f208bb7726",
 "metadata":{},
 "uploadType":"LOCAL",
 "projectId": null}
```

 * Run migrations
 * Bring up your server
 * Post the test json to `/api/uploads/` -- it should succeed, and the response should have `null`s for `sceneAcquisitionDate` and `sceneCloudCover` in `sceneMetadata`
 * Add a bogus field to `metadata` (`"bogus": "bogus"` is fine) and post it again -- the `bogus` field should end up in the response's `metadata`
 * Leave `bogus` in there, and add `"sceneCloudCover": 0.7` to `metadata` in the post json -- `sceneCloudCover` should end up in `sceneMetadata`, and `bogus` should still go into `metadata`
 * Add `"sceneAcquisitionDate": "2017-04-01T00:00:00Z"` to the post json -- it should succeed, and the two scene metadata fields should go to `sceneMetadata`, while `bogus` goes into `metadata`

Closes #2236
